extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/validation/ContractRequestValidator.java

```diff
@@ -40,4 +43,22 @@ public static Validator<JsonObject> instance() {
                 )
                 .build();
     }
+
+    private record MandatoryCounterPartyAddressOrConnectorAddress(JsonLdPath path) implements Validator<JsonObject> {
+
+        @Override
+        public ValidationResult validate(JsonObject input) {
```

Comment:
This can be simplified by checking if the first value is set and returning

Reply:
Rearranged the order to return as soon as `coutnerPartyAddress` is found.
Did the some for the `TransferRequestValidator` and `CatalogRequestValidator`.